File: app/src/main/java/com/example/fragment4/MainActivity.java
package com.example.fragment4;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = "MAINACTIVITY";
    FragmentManager fragmentManager;
    FragmentTransaction fragmentTransaction;
    Fragment1 fragment1;
    Fragment2 fragment2;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        fragment1=new Fragment1();
        fragment2=new Fragment2();
        Log.d(TAG,"oncreate");
    }

    public void add(View view) {
        if (!fragment1.isAdded()) {
            fragmentManager = getSupportFragmentManager();
            fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.add(R.id.frameLayout, fragment1, "fragment1");
            fragmentTransaction.addToBackStack(null);
            fragmentTransaction.commit();
        }

    }

    public void show(View view) {
        if(fragment1.isAdded()){
        fragmentManager = getSupportFragmentManager();
        fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.show(fragment1);
        fragmentTransaction.addToBackStack(null);
        fragmentTransaction.commit();}
        else{
            fragmentManager = getSupportFragmentManager();
            fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.show(fragment2);
            fragmentTransaction.addToBackStack(null);
            fragmentTransaction.commit();
        }
    }

    public void remove(View view) {
        if(fragment1.isAdded()){
            fragmentManager = getSupportFragmentManager();
            fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.remove(fragment1);
            fragmentTransaction.addToBackStack(null);
            fragmentTransaction.commit();}
        else{
            fragmentManager = getSupportFragmentManager();
            fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.remove(fragment2);
            fragmentTransaction.addToBackStack(null);
            fragmentTransaction.commit();
        }

    }

    public void replace(View view) {
        if(!fragment1.isAdded()){
            add(view);
        }
        else{
            fragmentManager = getSupportFragmentManager();
            fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.replace(R.id.frameLayout, fragment2, "fragment1");
            fragmentTransaction.addToBackStack(null);
            fragmentTransaction.commit();
        }

    }

    public void hide(View view) {
        if(fragment1.isAdded()){
        fragmentManager = getSupportFragmentManager();
        fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.hide(fragment1);
        fragmentTransaction.addToBackStack(null);
        fragmentTransaction.commit();}
        else{
            fragmentManager = getSupportFragmentManager();
            fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.hide(fragment2);
            fragmentTransaction.addToBackStack(null);
            fragmentTransaction.commit();
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(TAG,"onstart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG,"onresumne");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG,"onpause");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d(TAG,"onrestart");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(TAG,"ondestroy");
    }
}

File: app/src/main/java/com/example/gallery1/view/activity/MainActivity.kt
package com.example.gallery1.view.activity

import android.content.Context
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager
import com.example.gallery1.R
import com.example.view.gallery1.LoginFragment

class MainActivity : AppCompatActivity() {
    companion object {
        lateinit var manager: FragmentManager
        var instance: MainActivity? = null
        fun applicationContext(): Context {
            return instance!!.applicationContext
        }


    }

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowTitleEnabled(false)


        manager = this.supportFragmentManager
        val transaction = manager.beginTransaction()
        val login_fragment = LoginFragment()
        transaction.add(R.id.login_holder, login_fragment)
        /*  transaction.addToBackStack(null)*/
        transaction.commit()

    }

    fun getTitle1(titleFrag: String) {
        supportActionBar!!.title = titleFrag
    }

    override fun onBackPressed() {

        super.onBackPressed()
        /* var count =supportFragmentManager.backStackEntryCount
         Log.d("backpress", count.toString())
         if (count == 0) {
             super.onBackPressed()
             finish()
         } else {
             supportFragmentManager.popBackStack()
         }*/
    }

//    override fun onOptionsItemSelected(item: MenuItem): Boolean {
//        var id = item.itemId
//        if (id == R.id.login) {
//            supportActionBar?.title ="login"
//        }
//        return super.onOptionsItemSelected(item)
//
//    }

    override fun onSupportNavigateUp(): Boolean {
        Log.d("backpress", "top left back")
        this.onBackPressed()
        return super.onSupportNavigateUp()


    }

}
